### PR TITLE
Implements #18 Create initial History-Row for existing data

### PR DESCRIPTION
**Pull request by droid42**
Consolidate History ensures that a create-row exists for each entity in history table.
Use identical timestamp on create for entity and history

Implements #19 Improve "defer tx.Rollback()"

### CODE REVIEW

File: pkg/datastore/bootstrap.go [hunk 1]
```diff
@@ -4,36 +4,103 @@ import (
...
+		if err != nil {
+			ds.log.Error("error consolidate history", zap.Error(err))
+		}
+	}
+
 	healthServer.SetServingStatus("initdb", healthv1.HealthCheckResponse_SERVING)
 	return nil
 }
 
+// consolidateHistory ensures, that for each VersionedJSONEntity there is at least one "created"-row in the history table.
```
Comment (majst01) [inline pkg/datastore/bootstrap.go:46]: As far as i understood you problem statement, you would like to create a history entry for all entries in the current table if none exists, but iterating over the configuration file will not cope these entries which have been created via api call before the history functionality was implemented.
Wouldnt it be better to iterate over all of the the current entries and check if a history entry for `created` exists and create one if not ?

File: pkg/datastore/postgres.go [hunk 2]
```diff
@@ -144,12 +146,7 @@ func (ds *Datastore) Create(ctx context.Context, ve VersionedJSONEntity) error {
 	if err != nil {
 		return err
 	}
-	defer func() {
-		err := tx.Rollback() // The rollback will be ignored if the tx has been committed later in the function.
-		if err != nil {
-			ds.log.Error("error rolling back", zap.Error(err))
-		}
-	}()
+	defer ds.rollback(tx)
```
Comment (majst01) [inline pkg/datastore/postgres.go:149]: Better

File: pkg/datastore/bootstrap.go (hunk 1 above)
Comment (droid42) [reply pkg/datastore/bootstrap.go:46]: This is exactly what I do. The consolidation of the history table is done by checking the existing rows in the entity-tables for missing created-rows and creating them, there is no relation to the config-file processing, except that it is triggered in a func named initdb() and the comment above the func is now misleading.

Should I create another func "migrateDB" or something that makes it clear that the consolidation is something separate and introduce a separate servingStatus.

Comment (majst01) [reply pkg/datastore/bootstrap.go:46]: OK, got it, was confused by the changes to the listFiles call.
This approach will of course be always run, and is only required once, should we introduce a version table which is incremented on every major table related change and run this part of the code if version is smaller than required ?

Comment (droid42) [reply pkg/datastore/bootstrap.go:46]: I see your point, will try to implement a lean mechanism

Comment (droid42) [reply pkg/datastore/bootstrap.go]: Please take a look

File: pkg/datastore/migrate.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+		return err
+	}
+
+	healthServer.SetServingStatus("migratedb", healthv1.HealthCheckResponse_SERVING)
+	return nil
+}
+
+// consolidateHistory ensures, that for each VersionedJSONEntity there is at least one "created"-row in the history table.
+// The type of entities to consolidate is specified by the given pointer to a slice of entities.
+func (ds *Datastore) consolidateHistory(tx *sql.Tx, entitySlicePrt interface{}) error {
```
Comment (majst01) [inline pkg/datastore/migrate.go:56]: typo: entitySlicePrt -> entitySlicePtr

Comment (majst01) [reply pkg/datastore/migrate.go:56]: if fixed, i will merge

Comment (droid42) [reply pkg/datastore/migrate.go]: Fixed

**Pull request merged**